src/data-access/repositories/mongo/base-mongo-repository.ts

```diff
@@ -1,17 +1,17 @@
 import { MongoDatabase } from './mongo-database'
-import { MongoEntityConverter } from './mongo-entity-converter'
+import { MongoEntityConverter, MongoId } from './mongo-entity-converter'
 import { Collection } from 'mongodb'
 import { DatabaseNotConnectedException } from '../../../model/exceptions/database-not-connected-exception'
 
-export abstract class BaseMongoRepository {
+export abstract class BaseMongoRepository<T extends MongoId = MongoId> {
   protected constructor(
-    protected mongoDatabase: MongoDatabase,
+    protected mongoDatabase: MongoDatabase<T>,
```

Comment:
This seems like a wrong use of generics. When would you restrict the entire database to a single model type?
My guess is that you want to have the generic type applying to the result of `getCollection`. If that is the case, you can achieve by making the getCollection method generic, like so:
```typescript
  protected <Model extends MongoId>getCollection(): Collection<Model>
```
And it would be nice if we used more descriptive names for generics than single letters :smile:

Reply:
This generic is to tell MongoDB that we are not using `ObjectId` on our Collection. If I just set the generic on the method, MongoDB still thinks that we are using `ObjectId` which gives us compilation errors on MongoDB version 6.

If I place the generic only on the method I get the error `Collection<MongoId> is not assignable to Collection<T>`.

I'm open for solutions to this.

> And it would be nice if we used more descriptive names for generics than single letters

`T` is the standard the letter to use when declaring a generic in most languages. Notice that even the error message we got from MongoDB is using `T`.
But it does seem like that it is not the standard in Typescript :p

Reply:
Which version of Mongo are you using? I belive your suggestion does not give any errors on versions below version 6.

Reply:
> T is the standard the letter to use when declaring a generic in most languages. Notice that even the error message we got from MongoDB is using T.
Just because something is widely used, does not mean that it is a good idea :wink:

The mongo driver version that I use is the one specified in package.json on this branch.
I tried applying the suggested change on this branch locally, and I got no warnings nor errors when compiling or running the code.

Reply:
> Just because something is widely used, does not mean that it is a good idea

Notice my last sentece ;)

Reply:
Changed `T` to `Model`

Reply:
Merged your changes for this :)